package.json

```diff
@@ -2,13 +2,18 @@
   "name": "dids",
   "version": "3.0.0-alpha.0",
   "description": "Typescript library for interacting with DIDs",
-  "main": "lib/index.js",
+  "main": "./lib/index.js",
+  "type": "module",
+  "exports": {
+    ".": "./lib/index.js"
+  },
   "files": [
     "lib/*"
   ],
+  "sideEffects": false,
   "scripts": {
     "test": "jest",
-    "build": "tsc -p tsconfig.json",
+    "build": "./node_modules/.bin/tsc --project tsconfig.build.json",
```

Comment:
Is this needed to specify the path to `tsc`? My understanding is that npm handles this, no?

Reply:
`npm` handles this indeed by providing `$PATH` that (for npm!) starts with `./node_modules/.bin`. A user though could forget to `npm install`: global (and it is another question what is "global" based on `$PATH` provided by npm, yarn or pnpm) `tsc` will be used. Or, sometimes, you could decide to run a script without relying on npm `$PATH`. Then you have to convert path to the executable by yourself. I understand, that these are edge cases, but they are really painful, so unless you strongly oppose this change, I would like to go with `./node_modules/.bin/...`.

Reply:
I'm wondering how likely are these edge cases to happen? In CI this should never be the case unless there's a config issue, and locally I'd expect forgetting to run `npm install` would be problematic anyways.

I don't mind having this change if you feel strongly about it, I guess it's mostly that I'd like to have something consistent between scripts and repos, so here if we do this for `tsc` I think we should also do it for `eslint`, `jest` and any other, no?
It's a bit annoying though if we start moving to a monorepo (not in scope now but I think we could have this package, the key DID resolver and providers in the same repo), where I think we'd probably want to use the hoisted dependencies instead.

Reply:
Ooh, if moving to a monorepo is on the agenda (even in future), then using paths instead of command names is a necessity. I had a lot of confusion/mismatches in js-ceramic repository. Going to add paths to jest, eslint and others.

Just in case, agree on low expected probability of the edge cases.